feat: Install ACR in k3a cluster

vpatelsj opened the pull request.

vpatelsj committed:
feat: Install ACR in k3a cluster

jwilder commented on acr/secret.go:
```diff
@@ -0,0 +1,119 @@
+		"create", "secret", "docker-registry", secretName,
+		"--docker-server=" + loginServer,
+		"--docker-username=00000000-0000-0000-0000-000000000000",
+		"--docker-password=" + token,
```
This will expire.  msi-acr-pull rotates them automatically.

jwilder commented on cluster/create.go:
```diff
@@ -87,32 +110,32 @@ func createKeyVault(ctx context.Context, subscriptionID, cluster, location, vnet
 	certAdminRoleDefID := fmt.Sprintf("/subscriptions/%s/providers/Microsoft.Authorization/roleDefinitions/a4417e6f-fecd-4de8-b567-7b0420556985", subscriptionID)
 	certAdminRoleName := kstrings.DeterministicGUID(keyVaultID + msiPrincipalID + "a4417e6f-fecd-4de8-b567-7b0420556985")
-	if _, err = roleAssignmentsClient.Create(ctx, keyVaultID, certAdminRoleName, armauthorization.RoleAssignmentCreateParameters{
+	if err = assignRoleWithRetry(ctx, roleAssignmentsClient, keyVaultID, certAdminRoleName, armauthorization.RoleAssignmentCreateParameters{
```
Oh good.  This bug was annoying and just needed a retry.

jwilder commented on cluster/create.go:
```diff
@@ -472,13 +498,18 @@ func createVirtualNetwork(ctx context.Context, subscriptionID, resourceGroup, lo
 
+	// Use default SKU if none provided
+	if postgresSKU == "" {
+		postgresSKU = "Standard_D2s_v3"
```
This should come in via the default value of the flag to be consistent with the other options.

jwilder commented on cmd/k3a/acr.go:
```diff
@@ -0,0 +1,239 @@
+	createACRCmd.Flags().String("cluster", "", "Cluster name (or set K3A_CLUSTER) (required)")
+	createACRCmd.Flags().String("name", "", "ACR registry name (required)")
+	createACRCmd.Flags().String("region", "", "Azure region for the ACR (e.g., canadacentral) (required)")
+	createACRCmd.Flags().String("sku", "Basic", "ACR SKU (Basic, Standard, Premium)")
```
I'd use `acr-sku` instead of just `sku` to make this more clear.

jwilder commented on cmd/k3a/acr.go:
```diff
@@ -0,0 +1,239 @@
+		}
+		clusterName, _ := cmd.Flags().GetString("cluster")
+		if clusterName == "" {
+			return fmt.Errorf("--cluster flag is required (or set K3A_CLUSTER)")
```
Are these actually using the common `K3A_*` env vars?  Doesn't look like they are wired up in these commands.

vpatelsj committed:
Improve k3s cluster configuration

- Expand cluster CIDR to 10.0.0.0/8 with larger node CIDR masks (/22)
- Increase max pods per node to 500
- Remove centralized node labeling service for better scalability
- Use worker node self-labeling with --node-label flag
- Update VNet subnet ranges to align with larger CIDR

vpatelsj committed:
expand cluster cidr and disable node labeller

vpatelsj committed:
working large cluster cidr

vpatelsj committed:
kine table cleaner

vpatelsj committed:
add indexer

vpatelsj committed:
fix node count

vpatelsj committed:
vapa18 optimizations
